app/models/health/qualifying_activity_v2.rb

```diff
@@ -88,73 +88,124 @@ def self.activities
+          per_day: 1,
         },
         cha_completed: {
           title: 'Comprehensive Assessment completed',
           code: 'G0506>U4',
           weight: 15,
           hidden: true,
+          allowed: [],
+          required: ['U4'],
+          per_day: 1,
         },
         care_planning: {
           title: 'Development of Care Plan',
           code: 'T2024',
           weight: 20,
           hidden: true,
+          allowed: ['U1', 'UK', 'U2', 'U3', '95'],
+          required: [],
+          per_day: 3,
         },
         care_coordination: {
           title: 'Care coordination',
           code: 'G9005',
           weight: 30,
+          allowed: ['U1', 'UK', 'U2', 'U3', '95'],
+          required: [],
+          per_day: 4,
         },
         care_team: {
           title: 'Meeting with 3+ care team members',
           code: 'G9007',
           weight: 40,
+          allowed: ['U1', 'U2', 'U3', '95'],
+          required: [],
+          per_day: 1,
         },
         care_transitions: {
           title: 'Emergency Department visit (7 days)',
           code: 'T2038',
           weight: 45,
+          allowed: ['U1', 'U2', 'U3', '95'],
+          required: [],
+          per_day: 2,
         },
         discharge_follow_up: {
           title: 'Follow-up from inpatient discharge with client (7 days)',
           code: 'T2038>U5',
           weight: 50,
+          allowed: ['U1', 'U2'],
+          required: ['U5'],
+          per_day: 2,
         },
         pctp_signed: {
           title: 'Care Plan completed',
           code: 'T2024>U4',
           weight: 100,
           hidden: true,
+          allowed: [''],
+          required: ['U4'],
+          per_day: 1,
         },
         intake_completed: {
           title: 'Intake/reassessment (completing consent ROI, comprehensive assessment, care plan)',
           code: 'G9005',
           weight: 110,
+          allowed: ['U1', 'UK', 'U2', 'U3', '95'],
+          required: [],
+          per_day: 4,
         },
         sdoh_positive: {
           title: 'SDoH screening positive',
           code: 'G9919',
           weight: 200,
+          allowed: [],
+          required: [],
+          per_day: 1,
         },
         sdoh_negative: {
           title: 'SDoH screening negative',
           code: 'G9920',
           weight: 210,
+          allowed: [],
+          required: [],
+          per_day: 1,
         },
       }.sort_by { |_, m| m[:weight] }.to_h
     end
 
-    def contact_required?(activity)
-      return false unless activity
+    def internal_compute_valid_unpayable
+      reasons = []
+      computed_procedure_valid = @qa.compute_procedure_valid?
+
+      # Only valid procedures can be valid unpayable
+      return nil unless computed_procedure_valid
+
+      # Unpayable if it is a valid procedure, but it didn't occur during an enrollment
+      reasons << :outside_enrollment if computed_procedure_valid && ! @qa.occurred_during_any_enrollment?
```

Comment:
Same here, `computed_procedure_valid` is guarded on 195.
```suggestion
      reasons << :outside_enrollment unless @qa.occurred_during_any_enrollment?
```